sky/packages/sky/lib/rendering/object.dart

```diff
@@ -533,20 +660,39 @@ abstract class RenderObject extends AbstractNode implements HitTestTarget {
     markNeedsPaint();
     assert(parent == this.parent); // TODO(ianh): Remove this once the analyzer is cleverer
   }
-  bool get sizedByParent => false; // return true if the constraints are the only input to the sizing algorithm (in particular, child nodes have no impact)
-  void performResize(); // set the local dimensions, using only the constraints (only called if sizedByParent is true)
+
+  /// Whether the constraints are the only input to the sizing algorithm (in particular, child nodes have no impact)
+  ///
+  /// Returning false is always correct, but returning true can be more
+  /// efficient in some siutations.
+  bool get sizedByParent => false;
+
+  /// Updates the render objects size using only the constraints
+  ///
+  /// This function is called only if [sizedByParent] is true.
+  void performResize();
+
+  /// Do the work of computing the layout for this render object
+  ///
+  /// Do not call this function directly. This function is called by layout when
+  /// there is actually work to be done by this render object during layout. The
+  /// layout constraints provided by your parent are available via the
+  /// [constraints] getter.
+  ///
+  /// If [sizedByParent] is true, then this function should not actually change
+  /// the dimensions of this render object. Instead, that work should be done by
+  /// [performResize]. If [sizedByParent] is false, then this function should
+  /// both change the dimensions of this render object and instruct its children
+  /// to layout.
+  ///
+  /// When calling [layout] on your children, pass true for parentUsesSize if
+  /// your layout information is dependent on your child's layout information.
+  /// Passing true for parentUsesSize ensures that this render object will
+  /// undergo layout if the child undergoes layout. Otherwise, the child can
+  /// changes its layout information without informing this render object.
   void performLayout();
-    // Override this to perform relayout without your parent's
-    // involvement.
-    //
-    // This is called during layout. If sizedByParent is true, then
-    // performLayout() should not change your dimensions, only do that
-    // in performResize(). If sizedByParent is false, then set both
-    // your dimensions and do your children's layout here.
-    //
-    // When calling layout() on your children, pass in
-    // "parentUsesSize: true" if your size or layout is dependent on
-    // your child's size or intrinsic dimensions.
+
+  /// Updates state to account for re-entrant layout and invokes callback
```

Comment:
Technically the layout isn't re-entrant with this.

This just allow you to mutate your child list (and the rest of that subtree) during performLayout().